File: Practise/prac1.py
from typing import List

def nextPermutation(nums : List[int]) -> None:
    i : int = len(nums) - 2
    n : int = len(nums)
    
    while (i >= 0 and nums[i] >= nums[i+1]):
        i -= 1
    
    if i >= 0:
        swap : int = n - 1
        
        while(nums[i] >= nums[swap]):
            swap -= 1
        
        nums[i],nums[swap] = nums[swap],nums[i]
        
    nums[i+1:].reverse()

    if i == -1 :
       nums.sort() 

nums : List[int] = [3,2,1]
nextPermutation(nums)
print(nums)

File: Practise/prac1.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

void nextPermutation(vector<int>& nums, int start, vector<vector<int>>& permutations)
{
    if (start == nums.size() - 1) //
    {
        permutations.push_back(nums);
        return;
    }

    for (int i = start; i < nums.size(); i++)
    {
        swap(nums[i], nums[start]);
        nextPermutation(nums, start + 1, permutations);
        swap(nums[i], nums[start]);
    }
}

void generatePermutation(vector<int>& nums)
{
    vector<vector<int>> permutations;
    nextPermutation(nums, 0, permutations);

    
}